backend/app/adapter/db/search.go

```diff
@@ -0,0 +1,64 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"short/app/adapter/db/table"
+	"short/app/entity"
+	"short/app/usecase/search"
+)
+
+var _ search.Search = (*Search)(nil)
+
+// Search struct
+type Search struct {
+	db *sql.DB
+}
+
+// GetURLByType - to retrieve the url associated with userEmail based on type specified
+// TODO: Add urlType based querying after column addition
+func (urlSearch Search) GetURLByType(userEmail, urlType string) ([]search.URLSearchResponse, error) {
```

Comment:
Let's try `GetURLByVisibility` instead. `Type` doesn't seem to convey necessary information to readers.